elfi/methods/utils.py

```diff
@@ -184,29 +184,72 @@ def logpdf(cls, x, means, cov=1, weights=None):
         return np.log(cls.pdf(x, means=means, cov=cov, weights=weights))
 
     @classmethod
-    def rvs(cls, means, cov=1, weights=None, size=1, random_state=None):
+    def rvs(cls, means, cov=1, weights=None, size=1, prior_logpdf=None,
+            max_trials=100, random_state=None):
         """Draw random variates from the distribution.
 
         Parameters
         ----------
         means : array_like
             means of the Gaussian mixture components
-        weights : array_like
-            1d array of weights of the gaussian mixture components
-        cov : array_like
+        cov : array_like, optional
             a shared covariance matrix for the mixture components
-        size : int or tuple
+        weights : array_like, optional
+            1d array of weights of the gaussian mixture components
+        size : int or tuple, optional
+        prior_logpdf : callable, optional
+            Can be used to check validity of random variable.
+        max_trials : int, optional
```

Comment:
I think this is potentially somewhat dangerous in a very long running job. It would be very inconvenient if this value was accidentally exceeded and the whole job would be interrupted because of it.

Reply:
Yes, but what would be better? There has to be *some* limit, and my informal testing suggests less than 10 iterations are needed in our example models. 

In any case, the current implementation crashes very often, if the simulator is given bad parameter values and it can't handle it.

Reply:
I think it cannot be generally easily or at all set since it depends on the acceptance ratio which depends on many other factors (e.g. dimensionality). What if we would just remove this and put warnings to the logger if sampling takes a very long time?

Reply:
The crashing implementation is essentially a problem of the simulator. It could be fixed by the simulator checking it's input and returning an "empty" simulation for non suitable parameters.

Reply:
Of course changing SMC to provide only parameters that have positive support from the prior is a very convenient feature to have and reduces work from the simulator implementator.

Reply:
You mean as is (minus the arg) but instead of raising an exception just throw a warning? I guess that would be ok. The user can then kill the process it s/he wishes.